compose/animation/animation-core/src/commonMain/kotlin/androidx/compose/animation/core/CupertionScrollDecayAnimationSpec.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.animation.core
+
+import kotlin.math.abs
+import kotlin.math.ln
+import kotlin.math.pow
+
+/**
+ * A class that represents the animation specification for a scroll decay animation
+ * using iOS-style decay behavior.
+ *
+ * @property decelerationRate The rate at which the velocity decelerates over time.
+ * Default value is equal to one used by default UIScrollView behavior.
+ */
+class CupertinoScrollDecayAnimationSpec(
```

Comment:
I guess we should move `Cupertino*` classes to separate package. Also please note that it will not work on android without upstreaming

Reply:
Done